docs/other-api/dev.md

```diff
@@ -5,32 +5,100 @@ order: 1
+remix create ./my-app --template /path/to/remix-template
+remix create ./my-app --template /path/to/remix-template.tar.gz
+remix create ./my-app --template remix-run/grunge-stack
+remix create ./my-app --template :username/:repo
+remix create ./my-app --template https://github.com/:username/:repo
+remix create ./my-app --template https://github.com/:username/:repo/tree/:branch
+remix create ./my-app --template https://github.com/:username/:repo/archive/refs/tags/:tag.tar.gz
+remix create ./my-app --template https://example.com/remix-template.tar.gz
+```
+
+<aside aria-label="Private GitHub repo templates">
+<docs-info>
+
+To create a new project from a template in a private GitHub repo, set the `GITHUB_TOKEN` environment variable to a personal access token with access to that repo.
 
-Builds your app for production. No need to add `NODE_ENV=production` to the command.
+</docs-info>
+</aside>
+
+## `remix build`
+
+Builds your app for production. This command will set `process.env.NODE_ENV` to `production` and minify the output for deployment.
 
 ```sh
 remix build
 ```
 
-### `remix watch`
+### `remix build --sourcemap`
+
+Generates sourcemaps for the production build.
+
+## `remix watch`
 
-Watches your application files and builds your app for development when files change.
+Watches your application files and builds your app for development when changes are made.
 
 ```sh
 remix watch
 ```
 
-### `remix dev`
+## `remix dev`
 
-Same as `watch` but also boots the [Remix app server](serve.md) in development mode if it's installed.
+The same as `watch`, but also boots the [Remix App Server](serve.md) in development mode if it's installed.
 
 ```sh
 remix dev
 ```
 
-_Note: The default port is `3000`, but can be changed by setting the `PORT` environment variable. If you are changing this to run multiple instances of Remix, you must also set unique [Dev websocket ports](https://remix.run/docs/en/v1/api/conventions#devserverport) via `remix.config.js` or setting `REMIX_DEV_SERVER_WS_PORT`._
+### `remix dev --debug`
+
+Attaches a [Node inspector](https://nodejs.org/en/docs/guides/debugging-getting-started/) to develop your app in debug mode.
+
+### `remix dev --port`
+
+Launches the app server on a given port.
+
+By default, the port is set to `3000`. If port `3000` is unavailable, the `dev` command will attempt to find another port that is open. Using the `--port` flag will only attempt to launch the server at the given port; if the port is unavailable the app will not start.
+
+```sh
+remix dev --port 4001
+```
+
+Alternatively, a port can be assigned to the `PORT` environment variable.
+
+<aside aria-label="Running multiple instances of Remix">
+<docs-info>
+
+If you are changing the port to run multiple instances of Remix, you must also set unique [websocket ports](../api/conventions#devserverport) via `remix.config.js`, or by setting the `REMIX_DEV_SERVER_WS_PORT` environment variable.
+
+</docs-info>
+</aside>
```

Comment:
This should not be necessary anymore since we determine this dynamically and devs shouldn't have to worry about it anymore.

```suggestion

```